contracts/src/v0.8/functions/dev/1_0_0/Functions.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {CBOR, Buffer} from "../../../shared/vendor/solidity-cborutils/v2.0.0/CBOR.sol";
+
+/**
+ * @title Library for Chainlink Functions
+ */
+library Functions {
+  uint16 public constant REQUEST_DATA_VERSION = 1;
+  uint256 internal constant DEFAULT_BUFFER_SIZE = 256;
+
+  using CBOR for Buffer.buffer;
+
+  enum Location {
+    Inline,
+    Remote
+  }
+
+  enum CodeLanguage {
+    JavaScript
+    // In future version we may add other languages
+  }
+
+  struct Request {
+    Location codeLocation;
+    Location secretsLocation;
+    CodeLanguage language;
+    string source; // Source code for Location.Inline or url for Location.Remote
+    bytes secrets; // Encrypted secrets blob for Location.Inline or url for Location.Remote
+    string[] args;
```

Comment:
Do we want to add binary args now?

Reply:
I'll add it in the metadata PR (just so we can get this one through for Dev Ex)

Reply:
It should be a simple addition and might actually be relevant to Dev Ex. It can be called `bytesArgs` and would have type `[]bytes`.

  struct Request {
    Location codeLocation;
    Location secretsLocation; // Only Remote secrets are supported
    CodeLanguage language;
    string source; // Source code for Location.Inline or url for Location.Remote
    bytes encryptedSecretsReference; // URLS for Location.Remote secrets
    string[] args;
    bytes[] bytesArgs;
  }

Reply:
And also an additional `addBytesArgs` function similiar to `addArgs`